ds_import/flver_extractor.py

```diff
@@ -0,0 +1,370 @@
+from mesh import Mesh
+from vector2 import Vector2
+from vector3 import Vector3
+from binary_reader import BinaryReader
+from errors import UnreadableFormatError
+
+
+
+# TODO: add docstring comments throughout
+class FlverExtractor:
+
+    # define namedtuples
+    material_info = namedtuple("material_data", ["name_offset", "path_offset", "param_count", "param_start_index", "flags", "unknown1", "unknown2", "unknown3"])
+    mesh_info = namedtuple("mesh_info", ["is_dynamic", "material_index", "unknown1", "unknown2", "bone_index", "bone_index_count", "unknown3", "bone_indices_offset",
+                                        "faceset_count", "faceset_index_offset", "vertex_info_count", "vertex_info_offset"])
+    faceset_info = namedtuple("faceset_info", ["flags", "topology", "cull_backfaces", "unknown1", "unknown2",
+                                               "index_count", "buffer_offset", "buffer_size", "unknown3", "unknown4", "unknown5"] )
+    vertex_info = namedtuple("vertex_info", ["unknown1", "vertex_struct_formats_index", "per_vertex_size", "vertex_count",
+                                             "unknown2", "unknown3", "buffer_size", "buffer_offset"])
+    vertex_description_info = namedtuple("vertex_description_info", ["datatype_count", "unknown1", "unknown2", "description_offset"])
+    material_parameters = namedtuple("material_parameters", ["name_offset1", "name_offset2", "unknown1", "unknown2", "unknown3", "unknown4", "unknown5", "unknown6"])
+
+
+    __flver_file_path = ""
+    __metadata = None                   # flver file metadata-- data offset, number of meshes, bones, etc.
+
+    __materials_info = None             # per material: offsets to material names, material paths, parameter count, etc.
+    __mesh_info = None                  # per mesh: is_dynamic, material_index, bone_index, etc.
+    __faceset_info = None               # per faceset: information about each faceset buffer (indices count, offset to indices, etc.)
+    __vertex_info = None                # per vertex set: vertex_struct_formats_index, data size per vertex, vertices offset, vertex_count, etc.
+    __vertex_description_info = None    # per vertex description: count of datapoints in this description, offset to start of description
+    __material_parameters = None        # per material?: offsets to material/texture filenames(?)
+    __vertex_struct_formats = None      # list of ordered dictionaries describing the vertex data layouts
+
+    def extract_model(self):
+        """
+        Extracts the loaded Dark Souls flver file and returns as a Model
+
+        :return: A Dark Souls Model
+        """
+
+        if self.__flver_file_path == "":
+            raise Exception("A file must be loaded before extraction.  Use FlverExtractor.load_file(flver_filepath)")
+
+
+        # TODO: fix bug occurring when there is more than one faceset per mesh.  There is a "faceset_count" parameter stored in self.__mesh_info that probably should be referenced
```

Comment:
We (probably) wouldn't need this unless we want to at some point support different LOD models.  

Shouldn't be too bad a fix, so I say let's go for it.  I'll make an issue.

Reply:
It's not confirmed that this is a bug.  The original comment is outdated, and there is definitely some bug that exists when multiple facesets exist, but I'm not convinced it's happening here rather than when a model is converted to an obj.  I'll create an issue to investigate this further.

Reply:
Cool. remove the outdated todo?